tt_metal/impl/device/device_pool.cpp

```diff
@@ -250,6 +250,28 @@ void DevicePool::add_devices_to_pool(std::vector<chip_id_t> device_ids) {
     }
 }
 
+void DevicePool::register_worker_thread_for_device(Device* device, uint64_t worker_thread_id) {
+    if (this->worker_thread_ids_per_device.find(device)  != this->worker_thread_ids_per_device.end()) {
+        TT_FATAL(worker_thread_ids_per_device.at(device) == worker_thread_id, "Cannot register more than one worker thread per device.");;
```

Comment:
Ideally we should not be looking in the map twice, reusing the iterator returned by find.

Reply:
Done